extensions/subclasses/customcontext.py

```diff
@@ -96,37 +100,37 @@ def __init__(self, bot):
 
     async def my_on_message(self, message):
         """This function replaces `bot.on_message`"""
-        # Ignore bots without a heavy check on each command, this kills it right at the message receipt
-        if message.author.bot:
+        if not self._should_reply(message):
             return
-        # A check to see if the message should be ignored
-        for _ in range(1):
-            if str(message.author.id) in self.bot.c.trustedusers:
-                break
-            if message.guild is not None:
-                if str(message.guild.id) in self.bot.c.trustedservers:
-                    break
-                if str(message.guild.id) in self.bot.ignored.guilds:
-                    return
-                if str(message.channel.id) in self.bot.ignored.channels:
-                    return
-            if str(message.author.id) in self.bot.ignored.users:
-                return
-        # now we do the command processing
-        # this little bit allows the bot to process multiple commands in one message, separated by " && "
-        if " && " in message.content:
-            cmds = message.content.split(" && ")
-            for cmd in cmds:
-                message.content = cmd
-                ctx = await self.bot.get_context(message, cls=NoodleContext)
-                if ctx.valid:
-                    await self.bot.invoke(ctx)
-        else:
-            # otherwise, process as normal
-            ctx = await self.bot.get_context(message, cls=NoodleContext)
-            if ctx.valid:
-                await self.bot.invoke(ctx)
 
+        # process multiple commands in one message
+        # NB this prevents " && " from appearing in arguments.
+        # it may also be worth re.splitting on r'\s+&&\s+',
+        # which allows other kinds of whitespace, but reduces performance
+        commands = message.content.split(" && ")
+        for command in commands:
+            sub_message = copy.copy(message)  # prevents some unwanted state mutation
+            sub_message.content = command
+            await self.process_commands(sub_message)
+            del sub_message  # save some memory in case there's more to process
+
+     def _should_reply(self, message):
```

Comment:
I like what is done here, but currently as it stands, if a user is not trusted (a normal user) - and not in a trusted guild, the bot will ignore them.
It seems these don't catch for that case and don't return "should respond = true"

Reply:
My line 120 does allow that possibility, I believe.  I added that after your feedback on Discord, but did a force push instead of a separate commit.
I agree that there should be a default `return True` at line 126, to emulate the fall through behavior of a for loop.

Reply:
Ah sorry.

So to clarify. There's three "states" a user can be.
- ignored - the bot will ignore you no matter what
- trusted - more or less equivalent to a superuser/admin - able to bypass checks and ratelimits for the bot
- '''other''' - almost all users live here, they're not ignored, and not a trusted person. just a regular person. by default we dont want to ignore them or give them "trusted" status